### PR TITLE
Added basic usage documentation

### PR DESCRIPTION
**Pull request by trinopoty**

### CODE REVIEW

**Comment by trinopoty** [comment]
@darrachequesne can you please take a look at this.

File: README.md [hunk 1]
```diff
@@ -3,14 +3,100 @@
...
 
 ## Usage
 Usage is slightly different based on the server being used.
 
-TODO: Add usage for Jetty and Tomcat.
+Create a servlet to handle the HTTP requests as follows:
+```java
+public class EngineIoServlet extends HttpServlet {
+
+    private static final EngineIoServer ENGINE_IO_SERVER = new EngineIoServer();
```
Comment (darrachequesne) [inline README.md]: Is `static` mandatory here? Is it possible to create multiple servlet with the same underlying `EngineIoServer`?

File: README.md [hunk 2]
```diff
@@ -3,14 +3,100 @@
...
+    protected void service(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        ENGINE_IO_SERVER.handleRequest(request, response);
+    }
+}
+```
+
+Listen for new connections as follows:
+```java
+EngineIoServer server = EngineIoServlet.getEngineIoServer();
+server.on("connection", new Emitter.Listener() {
```
Comment (darrachequesne) [inline README.md]: Would the lambda version be more readable?

```java
server.on("connection", (args) -> {
  EngineIoSocket socket = (EngineIoSocket) args[0];
})
```

File: README.md [hunk 3]
```diff
@@ -3,14 +3,100 @@
...
+        // Do something with socket
+    }
+});
+```
+
+Listen for packets as follows:
+```java
+EngineIoSocket socket;
+
+socket.on("packet", new Emitter.Listener() {
```
Comment (darrachequesne) [inline README.md]: Could you please clarify the difference between `packet` and `message`?

File: README.md [hunk 4]
```diff
@@ -3,14 +3,100 @@
...
+    public void call(Object... args) {
+        EngineIoSocket socket = (EngineIoSocket) args[0];
+        // Do something with socket
+    }
+});
+```
+
+Listen for packets as follows:
+```java
+EngineIoSocket socket;
```
Comment (darrachequesne) [inline README.md]: Where does the `socket` object come from?

File: README.md (hunk 1 above)
Comment (trinopoty) [reply README.md]: Creating the servlet is left up to the user. Normally, only one instance of EngineIoServer should exist per servlet instance. Multiple instances are possible if the user so desires. In this example, I've set it to static to be able to access the EngineIoServer instance from outside the servlet.

File: README.md (hunk 2 above)
Comment (trinopoty) [reply README.md]: Since the min requirement is Java7, I've left it as anon class impl. Should I use the lambda instead?

File: README.md (hunk 4 above)
Comment (trinopoty) [reply README.md]: The socket object should come from the servlet or other place as defined by the user. I'll make it more clear.

File: README.md (hunk 1 above)
Comment (darrachequesne) [reply README.md]: Let's not encourage the reader to create a global variable then :+1:

File: README.md (hunk 2 above)
Comment (darrachequesne) [reply README.md]: It's more readable , don't you think?

As a side-note, running Java7 in production in 2018 should be prohibited, shoudn't it? :smile:

Comment (trinopoty) [reply README.md]: It should but that doesn't stop people from doing it.
Let's keep this as is for now. It'll be updated when Java 7 support is dropped.

**Pull request merged**